waiter/src/waiter/cookie_support.clj

```diff
@@ -19,35 +19,19 @@
             [taoensso.nippy :as nippy]
             [waiter.utils :as utils])
   (:import clojure.lang.ExceptionInfo
+           java.net.HttpCookie
            java.nio.charset.StandardCharsets
            org.eclipse.jetty.util.UrlEncoded))
 
-(defn url-decode
-  "Decode a URL-encoded string.  java.util.URLDecoder is super slow.  Also Jetty 9.3 adds an overload
-  to decodeString that takes just a string.  This implementation should use that once we upgrade."
-  [^String string]
-  (when string
-    (UrlEncoded/decodeString string 0 (count string) StandardCharsets/UTF_8)))
-
-(defn- strip-double-quotes
-  [value]
-  (let [value-length (count value)]
-    (if (and (> value-length 1) (str/starts-with? value "\"") (str/ends-with? value "\""))
-      (subs value 1 (dec value-length))
-      value)))
-
-(defn cookie-value
```

Comment:
Definitely a win for simplicity, but I'm concerned about performance regressions.